libbeat/docs/shared-instrumentation.asciidoc

```diff
@@ -0,0 +1,91 @@
+[[configuration-instrumentation]]
+== Configure APM instrumentation
+
+++++
+<titleabbrev>Instrumentation</titleabbrev>
+++++
+
+Libbeat uses the Elastic APM Go Agent to instrument its publishing pipeline.
+Currently, only the Elasticsearch output is instrumented.
```

Comment:
I wouldn't know how to put this, but worth mentioning that the Tracer is stored in the `Beat` object and it can be used in any beat to instrument their own code paths. Currently the one beat that actually does that is the APM Server itself.
There is no much that can users can do about this...